File: src/main/java/com/company/salesreact/entity/Product.java
package com.company.salesreact.entity;

import io.jmix.core.FileRef;
import io.jmix.core.entity.annotation.JmixGeneratedValue;
import io.jmix.core.metamodel.annotation.InstanceName;
import io.jmix.core.metamodel.annotation.JmixEntity;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.UUID;

@JmixEntity
@Table(name = "PRODUCT")
@Entity
public class Product {
    @JmixGeneratedValue
    @Column(name = "ID", nullable = false)
    @Id
    private UUID id;

    @InstanceName
    @Column(name = "NAME", nullable = false, unique = true)
    private String name;

    @Column(name = "SPECIAL")
    private Boolean special;

    @Column(name = "PRICE", precision = 19, scale = 2)
    private BigDecimal price;

    @Column(name = "PHOTO")
    @Lob
    private FileRef photo;

    public FileRef getPhoto() {
        return photo;
    }

    public void setPhoto(FileRef photo) {
        this.photo = photo;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public void setPrice(BigDecimal price) {
        this.price = price;
    }

    public Boolean getSpecial() {
        return special;
    }

    public void setSpecial(Boolean special) {
        this.special = special;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }
}

File: frontend/src/jmix/entities/Product.ts
export class Product {
  static NAME = "Product";
  id?: string;
  name?: string | null;
  special?: boolean | null;
  price?: any | null;
  photo?: any | null;
}
export type ProductViewName = "_base" | "_instance_name" | "_local";
export type ProductView<V extends ProductViewName> = V extends "_base"
  ? Pick<Product, "id" | "name" | "special" | "price" | "photo">
  : V extends "_instance_name"
  ? Pick<Product, "id" | "name">
  : V extends "_local"
  ? Pick<Product, "id" | "name" | "special" | "price" | "photo">
  : never;
